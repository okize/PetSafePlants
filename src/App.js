import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Container, Content, Button } from 'native-base';
import call from 'react-native-phone-call';

// ASPCA Animal Poison Control Center Phone Number
const POISON_HOTLINE = '8884264435';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: 50,
  },
  header: {
    marginBottom: 20,
  },
});

export default class App extends React.Component {
  handleOnPress = () => {
    const args = {
      number: POISON_HOTLINE,
      prompt: true,
    };

    // eslint-disable-next-line no-console
    call(args).catch(console.error);
  }

  render() {
    return (
      <Container style={styles.container}>
        <Content>
          <Text style={styles.header}>Pet Safe Plants</Text>
          <Button onPress={this.handleOnPress}>
            <Text>Call Animal Poison Control</Text>
          </Button>
        </Content>
      </Container>
    );
  }
}
