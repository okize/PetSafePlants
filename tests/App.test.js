import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

test('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});
