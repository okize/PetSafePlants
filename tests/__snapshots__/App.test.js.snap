// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Styled(Container)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 50,
    }
  }
>
  <Styled(Content)>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      Pet Safe Plants
    </Text>
    <Styled(Button)
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Call Animal Poison Control
      </Text>
    </Styled(Button)>
  </Styled(Content)>
</Styled(Container)>
`;
